### DISCUSSION TITLE
How to import private key of metamask to wallet-core?

### DISCUSSION THREAD

**Discussion by Hazhir007** [trustwallet/wallet-core#1625] [Q&A] [answered]
Hi, how to import a private key that has been exported from Metamask to wallet-core for ethereum ? in cpp
i read the whole source code did not find a method or a way to import a private key ,
i also got another question when i get the private key of the demo that is implemented in cpp the returned string is as follows :
"0x16b4710" which does not look like a real private key for ethereum , how to parse this to be as a standalone private key that can be imported to other wallets.

**Comment by hewigovens** [comment]
Use `PrivateKey` class directly

**Comment by Hazhir007** [comment]
> Use `PrivateKey` class directly

thanks can you provide an example? i read the PrivateKey.h can not understand how to use it , i got different errors even when try to validate a private key with isValid method it gives me that i have to use a TWcureve when i provide the cure then it tells me that the method only accepts one argument , this is how i call it:

TW::PrivateKey::isValid("2a55a49995cf26a86f19a5c9272ee04b5a9f401130e29854c3a22fd50b309522", TWCurveSECP256k1);

**Comment by optout21** [comment, marked as answer]
'PrivateKey::isValid' has a one-parameter version, and a two-paramter version, and the first parameter is the key data, but it must be `Data` type, not string.  Hint: use parse("") from HexCoding.
https://github.com/trustwallet/wallet-core/blob/master/src/PrivateKey.h#L33
Test:
https://github.com/trustwallet/wallet-core/blob/master/tests/PrivateKeyTests.cpp#L20
Test with create (TW):
https://github.com/trustwallet/wallet-core/blob/master/tests/interface/TWPrivateKeyTests.cpp#L21
